mdata/aggregator.go

```diff
@@ -99,6 +100,51 @@ func (agg *Aggregator) flush() {
 	agg.agg.Reset()
 }
 
+// flush returns the aggregated point according to consolidator, and resets aggregation state
+func flush(aggregation *Aggregation, consolidator consolidation.Consolidator, currentBoundary uint32) schema.Point {
+	aggregatedPoint := schema.Point{Val: 0, Ts: 0}
+	// aggMetric.getAggregated guarantees that consolidator should be one of the followings.
+	switch consolidator {
+	case consolidation.Cnt:
+		aggregatedPoint = schema.Point{Val: aggregation.Cnt, Ts: currentBoundary}
+	case consolidation.Lst:
+		aggregatedPoint = schema.Point{Val: aggregation.Lst, Ts: currentBoundary}
+	case consolidation.Min:
+		aggregatedPoint = schema.Point{Val: aggregation.Min, Ts: currentBoundary}
+	case consolidation.Max:
+		aggregatedPoint = schema.Point{Val: aggregation.Max, Ts: currentBoundary}
+	case consolidation.Sum:
+		aggregatedPoint = schema.Point{Val: aggregation.Sum, Ts: currentBoundary}
+	}
+	return aggregatedPoint
+}
+
+// GetFromRob duplicates the underlying Aggregation,
+// consumes all points from rob, and return the newly aggregated points.
+// GetFromRob won't actually add those points to AggMetric.
+func (agg *Aggregator) GetFromRob(consolidator consolidation.Consolidator, from, to uint32, rob *ReorderBuffer) []schema.Point {
```

Comment:
I think this function should just take a `[]schema.Point` rather than a `*ReorderBuffer`. Then this component doesn't need to know about ROB.